src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -239,21 +239,25 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
                 $lastUse = $use;
```

Comment:
Early feedback: personally I would add namespace analysis here:

```suggestion
                $lastUse = $use;
                
                if (true === $this->configuration['import_symbols']) {
                    // collect declarations of classes, interfaces, traits, enums (only for PHP 8.1+)
                    // collect relative usages (like `new DateTime()` in global namespace)
                }
```

probably I would change `$uses` to something like `array<string, class-string> $shortNamesMap` (like `baz => \Foo\Bar\Baz`), then pass it around like currently `$uses` and would extend with new found symbols like it's done now, but only if short name is not colliding with any other short name.

I believe this way the changes would be less invasive (no `bool $analyseRelativeUses` added in multiple places).

Reply:
done